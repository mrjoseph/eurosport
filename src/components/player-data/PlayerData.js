import WinLosses from '../win-losses/WinLosses';
import React from 'react';
import insertDecimal from '../../utils/insertDecimal';
import './playerData.css';

const calculateWeight = weight => weight / 1000;

const PlayerData = ({ rank, points, weight, height, age, last }) => {
  return(
      <div className="player-data-container">
        <div className="player-data-row"><span>rank:</span> {rank}</div>
        <div className="player-data-row"><span>points:</span> {points}</div>
        <div className="player-data-row"><span>weight:</span> {`${calculateWeight(weight)} kg`}</div>
        <div className="player-data-row"><span>height:</span> {`${insertDecimal(height)} m`}</div>
        <div className="player-data-row"><span>age:</span> {`${age} years old`}</div>
        <WinLosses score={last} />
      </div>
  );
};

export default PlayerData;
