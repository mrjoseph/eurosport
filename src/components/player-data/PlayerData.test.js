import React from 'react';
import renderer from 'react-test-renderer';
import PlayerData from './PlayerData';
describe('PlayerData', () => {
  it('renders wins and losses component ', () => {
    const data = {
      rank: 21,
      points: 1784,
      weight: 81000,
      height: 183,
      age: 33,
      last: [1, 1, 1, 0, 1]
    };

    const rendered = renderer.create(
        <PlayerData {...data} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
