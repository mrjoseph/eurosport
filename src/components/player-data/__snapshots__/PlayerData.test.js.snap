// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerData renders wins and losses component  1`] = `
<div
  className="player-data-container"
>
  <div
    className="player-data-row"
  >
    <span>
      rank:
    </span>
     
    21
  </div>
  <div
    className="player-data-row"
  >
    <span>
      points:
    </span>
     
    1784
  </div>
  <div
    className="player-data-row"
  >
    <span>
      weight:
    </span>
     
    81 kg
  </div>
  <div
    className="player-data-row"
  >
    <span>
      height:
    </span>
     
    1.83 m
  </div>
  <div
    className="player-data-row"
  >
    <span>
      age:
    </span>
     
    33 years old
  </div>
  <div
    className="win-loss"
  >
    <div>
      Wins: 
      4
    </div>
    <div>
      Losses: 
      1
    </div>
  </div>
</div>
`;
